nexus/src/app/background/sync_service_zone_nat.rs

```diff
@@ -0,0 +1,233 @@
+                                    continue;
+                                }
+                            };
+
+                            let external_address =
+                                ipnetwork::Ipv4Network::new(external_ip, 32)
+                                    .unwrap();
+
+                            let nat_value = Ipv4NatValues {
+                                external_address: nexus_db_model::Ipv4Net(
+                                    omicron_common::api::external::Ipv4Net(
+                                        external_address,
+                                    ),
+                                ),
+                                first_port: snat_cfg.first_port.into(),
+                                last_port: snat_cfg.last_port.into(),
+                                sled_address: sled_address.into(),
+                                vni: nexus_db_model::Vni(nic.vni.into()),
+                                mac: nexus_db_model::MacAddr(nic.mac.into()),
+                            };
+
+                            // Append ipv4 nat entry
+                            ipv4_nat_values.push(nat_value);
+                        }
+                        OmicronZoneType::Nexus { nic, external_ip, .. } => {
+                            let external_ip = match external_ip {
+                                IpAddr::V4(addr) => addr,
+                                IpAddr::V6(_) => {
+                                    warn!(
+                                        &log,
+                                        "ipv6 addresses for service zone nat not implemented";
+                                    );
+                                    continue;
+                                }
+                            };
+
+                            let external_address =
+                                ipnetwork::Ipv4Network::new(external_ip, 32)
+                                    .unwrap();
+
+                            let nat_value = Ipv4NatValues {
+                                external_address: nexus_db_model::Ipv4Net(
+                                    omicron_common::api::external::Ipv4Net(
+                                        external_address,
+                                    ),
+                                ),
+                                first_port: 0.into(),
+                                last_port: 65535.into(),
+                                sled_address: sled_address.into(),
+                                vni: nexus_db_model::Vni(nic.vni.into()),
+                                mac: nexus_db_model::MacAddr(nic.mac.into()),
+                            };
+
+                            // Append ipv4 nat entry
+                            ipv4_nat_values.push(nat_value);
+                        }
+
+                        _ => continue,
+                    }
+                }
+            }
+
+            if ipv4_nat_values.is_empty() {
```

Comment:
If I understand things correctly, as of today this should never be empty because Nexus must be running in order to execute the background task. If Nexus is running, we should have some entries here.

Reply:
I probably need to wordsmith the error message to indicate this more clearly